app/src/main/java/com/uw/fydp/flexeat/flexeat/MenuActivity.java

```diff
@@ -118,26 +136,19 @@ public void onSubmitOrderClick(View view) {
     @Override
     public void getSelectedItems() {
         for(int i = 0; i<adapter.getCount(); i++){
-            Fragment currentFragment = adapter.getFragment(i);
-
-            if(currentFragment instanceof AppetizerFragment){
-                AppetizerFragment temp = (AppetizerFragment) currentFragment;
-                for(int j = 0 ; j < temp.adapter.getCount(); j++){
-                    MenuItem currentItem = temp.adapter.getItem(j);
-                    if(currentItem.isCheck){
-                        selectedItems.add(currentItem);
-                    }
-                }
-            }else if(currentFragment instanceof MainCourseFragment){
-                MainCourseFragment temp = (MainCourseFragment) currentFragment;
-                for(int j = 0 ; j < temp.adapter.getCount(); j++){
-                    MenuItem currentItem = temp.adapter.getItem(j);
+            GenericFragment currentFragment = (GenericFragment) adapter.getFragment(i);
```

Comment:
are we going to be using a generic fragment for all of our activities? how adaptable is it?

Reply:
at least for the menu page this is a generic fragment. I will actually rename it to generic menu fragment